src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -19,15 +19,11 @@ namespace Microsoft.PowerShell
 {
     internal class CommandLineParameterParser
     {
-        internal CommandLineParameterParser(ConsoleHost p, Version ver, string bannerText, string helpText)
+        internal CommandLineParameterParser(ConsoleHost p, string bannerText, string helpText)
         {
-            Dbg.Assert(p != null, "parent ConsoleHost must be supplied");
-
             _bannerText = bannerText;
             _helpText = helpText;
-            _parent = p;
-            _ui = (ConsoleHostUserInterface)p.UI;
-            _ver = ver;
+            _ui = (p != null) ? (ConsoleHostUserInterface)p.UI : null;
```

Comment:
For better future-proofing, maybe we should provide a dummy `ConsoleHostUserInterface` implementation that does nothing. This would also avoid many of the other changes.